optuna/storages/rdb/storage.py

```diff
@@ -336,82 +338,94 @@ def get_trial_system_attrs(self, trial_id):
+                    )
+                    .filter(models.TrialParamModel.trial_id == best_trial.trial_id)
+                    .all()
+                )
+                param_dict = {}
+                param_distributions = {}
+                for param in params:
+                    distribution = distributions.json_to_distribution(param.distribution_json)
+                    param_dict[param.param_name] = distribution.to_external_repr(param.param_value)
+                    param_distributions[param.param_name] = distribution
+                user_attrs = session.query(models.TrialUserAttributeModel).filter(
+                    models.TrialUserAttributeModel.trial_id == best_trial.trial_id
+                )
+                system_attrs = session.query(models.TrialSystemAttributeModel).filter(
+                    models.TrialSystemAttributeModel.trial_id == best_trial.trial_id
+                )
+                intermediate = session.query(models.TrialValueModel).filter(
+                    models.TrialValueModel.trial_id == best_trial.trial_id
+                )
+                best_trial_frozen = structs.FrozenTrial(
+                    best_trial.number,
+                    structs.TrialState.COMPLETE,
+                    best_trial.value,
+                    best_trial.datetime_start,
+                    best_trial.datetime_complete,
+                    param_dict,
+                    param_distributions,
+                    {i.key: json.loads(i.value_json) for i in user_attrs},
+                    {i.key: json.loads(i.value_json) for i in system_attrs},
+                    # Use Value-Models here
+                    {value.step: value.value for value in intermediate},
+                    best_trial.trial_id,
+                )
+            user_attrs = session.query(models.StudyUserAttributeModel).filter(
+                models.StudyUserAttributeModel.study_id == study.study_id
+            )
+            system_attrs = session.query(models.StudySystemAttributeModel).filter(
+                models.StudySystemAttributeModel.study_id == study.study_id
             )
-            system_attrs = {attr.key: json.loads(attr.value_json) for attr in attributes}
-
-            # Consolidate StudySummary.
             study_summaries.append(
                 StudySummary(
-                    study_name=study_model.study_name,
-                    direction=self.get_study_direction(study_model.study_id),
-                    best_trial=best_trial,
-                    user_attrs=self.get_study_user_attrs(study_model.study_id),
-                    system_attrs=system_attrs,
-                    n_trials=len(study_trial_models),
-                    datetime_start=datetime_start,
-                    study_id=study_model.study_id,
+                    study_name=study.study_name,
+                    direction=study.direction,
+                    best_trial=best_trial_frozen,
```

Comment:
Though we basically want to keep the compatibility, putting `best_trial` seems to make the logic a bit complicated..
It may worth reconsidering the fields of `StudySummary`, where `best_trial` costs more than advantage in my opinion.

Reply:
Agree. For users who need the best trial information, they could easily retrieve it by executing `load_study(storage=URL, study=study.name).best_trial`.

Reply:
I'd like to defer the discussion to other PRs, but I agree with the opinion.